### PR TITLE
Server port can now be zero

### PR DESCRIPTION
**Pull request by rickardoberg**
Allow server port to be set to zero, for random assignment. Actual port used is logged so that tools can find out what it was.

### CODE REVIEW

File: community/server/src/main/java/org/neo4j/server/web/Jetty9WebServer.java [hunk 1]
```diff
@@ -373,7 +380,10 @@ protected void startJetty()
     {
         try
         {
+            System.out.println("Bar");
+            console.log( "Foo");
```
Comment (chrisvest) [inline community/server/src/main/java/org/neo4j/server/web/Jetty9WebServer.java]: This doesn't look right.

File: community/server/src/test/java/org/neo4j/server/web/TestJetty9WebServer.java [hunk 2]
```diff
@@ -42,16 +45,26 @@
...
 	public void shouldBeAbleToRestart() throws Throwable
 	{
-		// TODO: This is needed because WebServer has a cyclic
-		// dependency to NeoServer, which should be removed.
-		// Once that is done, we should instantiate WebServer
-		// here directly.
-		WrappingNeoServer neoServer = new WrappingNeoServer(dbRule.getGraphDatabaseAPI());
-		WebServer server = neoServer.getWebServer();
-
+        Jetty9WebServer server = new Jetty9WebServer( new SystemOutLogging() );
```
Comment (chrisvest) [inline community/server/src/test/java/org/neo4j/server/web/TestJetty9WebServer.java]: The indentation looks off.

**Comment by chrisvest** [comment]
Perhaps mention this in the manual?

**Comment by cleishm** [comment]
Replaced by #4240

**Pull request closed without merging**